classes/submission/Representation.inc.php

```diff
@@ -85,6 +85,33 @@ function getSubmissionId() {
 	}
 
 	/**
+	 * Set submission revision.
+	 * @param $submissionRevision int
+	 */
+	function setSubmissionRevision($submissionRevision) {
+		$this->setData('submissionRevision', $submissionRevision);
+	}
+
+	/**
+	 * Get submission revision.
+	 * @return int
+	 */
+	function getSubmissionRevision() {
+		return $this->getData('submissionRevision');
+	}
+
+	/**
+	 * Get the current revision id for a galley from submission
+	 * @param $contextId int
+	 * @return int
+	 */
+	function getCurrentVersionId($contextId = null) {
```

Comment:
Hmm, I wonder whether relying on this to fetch the current revision is going to be too expensive in performance. I think it'll be possible to have a fairly standard fetch function that supports revisions e.g. like this:

    function getActiveSubmissions($contextId, ..., $revision = REVISION_LATEST) {
    }

...where `define('REVISION_LATEST', 0);`

Then I'd suggest adding a column e.g. to the `articles` table called `current_revision`. This would be really fast and easy to fetch and wouldn't impose much maintenance overhead.